adapters/src/main/kotlin/com/krzykrucz/elesson/currentlesson/topic/usecase/ChooseTopicDtoHandler.kt

```diff
@@ -1,2 +1,37 @@
 package com.krzykrucz.elesson.currentlesson.topic.usecase
 
+import arrow.core.Either
+import arrow.core.extensions.either.traverse.sequence
+import arrow.core.fix
+import arrow.fx.IO
+import arrow.fx.extensions.fx
+import arrow.fx.extensions.io.applicative.applicative
+import arrow.fx.fix
+import com.krzykrucz.elesson.currentlesson.topic.ChooseTopicDto
+import com.krzykrucz.elesson.currentlesson.topic.domain.ChooseTopicError
+import com.krzykrucz.elesson.currentlesson.topic.domain.chooseTopic
+import com.krzykrucz.elesson.currentlesson.topic.usecase.persistence.checkIfAttendanceIsChecked
+import com.krzykrucz.elesson.currentlesson.topic.usecase.persistence.fetchFinishedLessonsCount
+import com.krzykrucz.elesson.currentlesson.topic.usecase.persistence.persistInProgressLesson
+import java.time.LocalDate
+
+
+fun handleChooseTopicDto(): (ChooseTopicDto) -> IO<Either<ChooseTopicError, Unit>> = { chooseTopicDto ->
```

Comment:
it's tested on the REST adapter layer, the question is if we need test on the `use case` layer?